### PR TITLE
perf: support simple aggregate API

### PR DESCRIPTION
**Pull request by huhushirley**
perf: support simple aggregate API

### CODE REVIEW

File: src/bll/mongodb-collection-engine/record-query.ts [hunk 1]
```diff
@@ -58,12 +70,81 @@ export class MongodbCollectionRecordQueryBllImpl implements RecordQueryBll<any,
...
+
+    const cursor = this.collection.aggregate(pipeline, aggOption)
+
+    // add readPreference for aggregate
+    if (options?.readPreference) cursor.withReadPreference(options.readPreference)
+
+    return this.stream(cursor)
+  }
+
+  formatSort(sort = {}) {
```
Comment (orangemi) [inline src/bll/mongodb-collection-engine/record-query.ts:118]: `sort` should be more spefic contructed.

And the process should be for streamable...

File: src/bll/mongodb-collection-engine/record-query.ts [hunk 2]
```diff
@@ -32,15 +42,28 @@ export class MongodbCollectionRecordQueryBllImpl implements RecordQueryBll<any,
       spaceId,
       entityId,
     }, conds)
+
+    const addFields = {}
+    const singleSort = {}
+    Object.keys(sort || {}).map(key => {
```
Comment (orangemi) [inline src/bll/mongodb-collection-engine/record-query.ts:48]: use `Object.entries(sort)` instead

File: src/bll/mongodb-collection-engine/util.ts [hunk 3]
```diff
@@ -127,7 +128,26 @@ export function decodeBsonQuery(query: Record<string, any> = {}): any {
 
     // special field value format for _id
     if (field === '_id') value = new ObjectId(value)
-    if (op === '$like') op = '$regex'
+    if (op === '$like') {
+      result.$and.push({
+        [field]: { 
+          $regex: value,
+          $options: '$i'
```
Comment (orangemi) [inline src/bll/mongodb-collection-engine/util.ts:135]: options should be `i` not `$i`

File: src/bll/mongodb-collection-engine/record-query.ts (hunk 2 above)
Comment (huhushirley) [reply src/bll/mongodb-collection-engine/record-query.ts]: done

File: src/bll/mongodb-collection-engine/util.ts (hunk 3 above)
Comment (huhushirley) [reply src/bll/mongodb-collection-engine/util.ts]: done

**Review by orangemi — APPROVED** [review]

**Pull request merged**